### PR TITLE
103 add course to new branche

### PR DESCRIPTION
**Pull request by mdomdo0**

### CODE REVIEW

**Comment by marektavac** [comment]
OK, but is it possible to start the name of pull request with #103 and not just 103?
And some description to pull request comment for reviewer (if is different from issue) to help him to create the review

File: WebAPI/Controllers/StudentController.cs [hunk 1]
```diff
@@ -180,5 +179,61 @@ public async Task<IActionResult> AddNewBlock([FromBody]AddNewBlockModel newBlock
...
+            }
+            else
+            {
+                block.CourseId = course.Id;
+                block.Day = (Day)timetableBlock.Day;
+                block.StartHour = (byte)timetableBlock.StartBlock;
+                block.Duration = (byte)(timetableBlock.EndBlock - timetableBlock.StartBlock);
+                block.Room = timetableBlock.Room;
+                block.Teacher = timetableBlock.Teacher;
+                block.BlockType = (BlockType)timetableBlock.Type;
```
Comment (marektavac) [inline WebAPI/Controllers/StudentController.cs]: The same code 2x... What about a method with param timetableBlock to copy data to block?

File: WebAPI/Controllers/StudentController.cs [hunk 2]
```diff
@@ -180,5 +179,61 @@ public async Task<IActionResult> AddNewBlock([FromBody]AddNewBlockModel newBlock
             }
             return Ok();
         }
+
+        [HttpPut("editBlock")]
+        public async Task<IActionResult> EditBlock([FromBody]AddNewBlockModel newBlockModel)
```
Comment (marektavac) [inline WebAPI/Controllers/StudentController.cs]: Question is (check other controllers) if this is not too much logic for controller...
If create course... shouldn't be part of the model...

File: WebAPI/Controllers/StudentController.cs [hunk 3]
```diff
@@ -180,5 +179,61 @@ public async Task<IActionResult> AddNewBlock([FromBody]AddNewBlockModel newBlock
...
+                block.CourseId = course.Id;
+                block.Day = (Day)timetableBlock.Day;
+                block.StartHour = (byte)timetableBlock.StartBlock;
+                block.Duration = (byte)(timetableBlock.EndBlock - timetableBlock.StartBlock);
+                block.Room = timetableBlock.Room;
+                block.Teacher = timetableBlock.Teacher;
+                block.BlockType = (BlockType)timetableBlock.Type;
+
+                await _courseService.AddAsync(course);
+                course.Timetable.AllBlocks.Add(block);
```
Comment (marektavac) [inline WebAPI/Controllers/StudentController.cs]: ``course.Timetable.AllBlocks.Add(block);`` - if student change block in his timetable, we want to change also the course?
Anyway you do not save this course with this block (save is earlier)
If new course is find, its whole timetable should be requested from API and added to our db...

File: WebAPI/Controllers/StudentController.cs [hunk 4]
```diff
@@ -180,5 +179,61 @@ public async Task<IActionResult> AddNewBlock([FromBody]AddNewBlockModel newBlock
...
+
+            if (student.Timetable == null)
+            {
+                return ErrorResponse($"Timetable for student with id: {student.Id} does not exist.");
+            }
+
+            Block block = new Block();
+            TimetableBlock timetableBlock = newBlockModel.TimetableBlock;
+            Course course = await _courseService.FindByNameAsync(timetableBlock.CourseName);
+            if (course == null)
```
Comment (marektavac) [inline WebAPI/Controllers/StudentController.cs]: inside this if, should be just:
1. new course
2. AddAsync(course)
else is not needed and 211-221 could be deleted

**Comment by marektavac** [comment]
@mdomdo0 you don't ask for review, that is why I done two reviews :-)

File: WebAPI/Controllers/StudentController.cs (hunk 4 above)
Comment (marektavac) [reply WebAPI/Controllers/StudentController.cs]: May be som model should be called responsible for adding new course (it will also call TAPI)

**Comment by marektavac** [comment]
@mdomdo0 este by som mal jednu otazku, novy kurz by sa predsa nemal dat vytvarat, tie sa naimportuju zo vzdelavania, ci nie?

**Comment by mdomdo0** [comment]
@marektavac je to pre pripad kedy dany predmet nieje v rozvrhu

**Comment by martingranak** [comment]
@marektavac vytvoril som branchu so zlym nazvom (add course to new BRANCHE) a mergol som ju so zlou branchou (allow user to edit existing block). 
Aby som to opravil tak som vytvoril novu branchu s dobrym nazvom mergol som ju s aktualnymi zmenami a vratil som zmeny, ktore boli v allow user to edit existing block aby tam neskor nevznikali konflikty.

**Comment by marektavac** [comment]
We can discuss this tomorrow, is it OK?

File: WebAPI/Controllers/StudentController.cs (hunk 1 above)
Comment (dobi28) [reply WebAPI/Controllers/StudentController.cs]: I resolved this in " Edit block on backend is done " commit at branch #79

**Pull request closed without merging**